pkg/virt-controller/services/template.go

```diff
@@ -130,8 +130,8 @@ type targetAnnotationsGenerator interface {
 }
 
 type TemplateService interface {
-	RenderMigrationManifest(vmi *v1.VirtualMachineInstance, sourcePod *k8sv1.Pod) (*k8sv1.Pod, error)
-	RenderLaunchManifest(vmi *v1.VirtualMachineInstance) (*k8sv1.Pod, error)
+	RenderMigrationManifest(vmi *v1.VirtualMachineInstance, sourcePod *k8sv1.Pod, backendStoragePVCName string) (*k8sv1.Pod, error)
+	RenderLaunchManifest(vmi *v1.VirtualMachineInstance, backendStoragePVCName string) (*k8sv1.Pod, error)
```

Comment:
What about using the pvcStore in the `templateService` to list the PVC and fetch the backendStorage PVC (if any)?

This would allow you to remove this additional parameter, I think it'd look better and would save a lot of lines of code that are just churn.

To achieve this I'd change the `withBackendStorage()` function signature to take a vmi and a cache.Store.

Reply:
> change the `withBackendStorage()` function signature to take a vmi and a cache.Store.

To me, that introduces a lot of potential for problems, and it would need a lot of code to answer some tricky questions, like:
What if the store hasn't updated yet? Which of the 2 PVCs is the target one in the case of the migration? Or is it a shared PVC situation? Is this manifest actually a migration target one or just a VMI start?

I think the backend storage PVC is as relevant as the source pod (I initially passed a pointer to the PVC object, like for `sourcePod`, but then realized only the name was used).

I can look into it if you feel strongly about it though...

Reply:
For the store update you can just use an expectation and make sure it is satisfied before calling this function, you actually have this racy behaviour in a few points.

Reply:
> For the store update you can just use an expectation and make sure it is satisfied before calling this function

That sounds great, but can you please develop? Where would we set and lift the expectation?

> For the migration part, I guess that you do this differentiation somewhere right?

Yes, `CreateIfNeededAndUpdateVolumeStatus` has an explicit migration parameter, true when called from migration.go, false when called from vmi.go.
For `withBackendStorage` to have the info however, we'd pretty much have to replace every PVC name param with a migration bool and, critically, add a migration param to `renderLaunchManifest()`, which we used to have and proudly removed IIRC...

> We use 2 different functions for creating the manifests

Not really, that's `renderLaunchManifest()` in both cases

> you might have to add some others parameters to withBackendStorage

It's not just a matter of params, it needs a way to find the name of the target PVC we just created in the case of a migration. Maybe we can store it in the VMI though.